components/track/TrackShareDialog.vue

```diff
@@ -0,0 +1,93 @@
+<script lang="ts" setup>
+// import type { NuxtIconName } from '.nuxt/nuxt-icons';
+
+const { t } = useI18n();
+
+const props = defineProps<{
+  trackId: number;
+}>();
+
+const emit = defineEmits<{
+  close: [];
+}>();
+
+const message = ref<string | null>(null);
+
+// TODO: Make this dynamic
```

Comment:
While making this dynamic, keep in mind that this has to work for Electron as well. This means, we'll need a fix value, at least for the domain.